### DISCUSSION TITLE
Workaround for brew install gcc: postinstall broken (gcc-13)

### DISCUSSION THREAD

**Discussion by jaredmdobson** [Homebrew/discussions#5256] [Linux] [answered]
### Additional Info

- [X] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.16
ORIGIN: https://github.com/Homebrew/brew
HEAD: bd1155be8f50998429a795c15a69c8fe75250510
Last commit: 28 hours ago
Core tap JSON: 02 Apr 19:44 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 3.1.4 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 16-core 64-bit zen3
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.146.1-microsoft-standard-WSL2 x86_64 GNU/Linux
OS: Ubuntu 22.04.4 LTS (jammy)
WSL: 2 (Microsoft Store)
Host glibc: 2.35
/usr/bin/gcc: 11.4.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: 13.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I believe related to: https://github.com/Homebrew/homebrew-core/issues/137991

```
Warning: The post-install step did not complete successfully
You can try again using:
  brew postinstall gcc
==> An exception occurred within a child process:
  ErrorDuringExecution: Failure while executing; `/usr/bin/cc -print-file-name=crti.o` exited with 127. Here's the output:
```

I was able to get it to work by install the build-essential package in ubuntu.

```bash
sudo apt install build-essential -y
```

I just thought that might be useful for anyone trying to install gcc-13 via homebrew on ubuntu wsl.

**Comment by ZhongRuoyu** [comment, marked as answer]
The problem you had is not related to the gcc issue you linked.

The `build-essential` package, which provides essential build tools, is needed by some of Homebrew's functionalities (see https://docs.brew.sh/Homebrew-on-Linux#requirements). The error you encountered is just an indication of it being missing.
